weave-js/src/components/Panel2/ChildPanel.tsx

```diff
@@ -147,6 +147,7 @@ export const initPanel = async (
   const {curPanelId: id} = getPanelStacksForType(inputNode.type, panelId, {
     allowedPanels,
     stackIdFilter: allowPanel,
+    excludePanelPanel: stack.length > 0,
```

Comment:
for this and the other similar line, can you do the following:
1. create a helper function `shouldExcludePanelPanel` that accepts the stack
2. add comments on that function detailing what is happening and why the hack